app.rb

```diff
@@ -1,14 +1,62 @@
+require 'json'
 require_relative 'teacher'
 require_relative 'student'
 require_relative 'book'
 require_relative 'rental'
+require 'pry'
 
 class App
   attr_accessor :books, :people
 
   def initialize
     @books = []
     @people = []
+    load_files
+  end
+
+  def load_files
+    books_file = File.open('books.json')
+    rental_file = File.open('rentals.json')
+    rentals = JSON.parse(rental_file.read, create_additions: true)
+    json = books_file.read
+    # Load books
+    if json != ''
+      arr = JSON.parse(json, create_additions: true)
+      arr.each { |books_json| @books << JSON.parse(books_json, create_additions: true) }
+    end
+    books_file.close
+    # Load people
+    people_file = File.open('people.json')
+    json = people_file.read
+    if json != ''
+      arr = JSON.parse(json, create_additions: true)
+      arr.each { |people_json| @people << JSON.parse(people_json, create_additions: true) }
+    end
+    # Load rentals
+    if rentals != ''
+      (0...rentals.length).each do |i|
+        rental = JSON.parse(rentals[i], create_additions: true)
+        @books.each { |book| book.rental << rental[0] if rental[0].book.title == book.title }
+        @people.each { |person| person.rental << rental[0] if rental[0].person.name == person.name }
+      end
+    end
+    # binding.pry
+    people_file.close
+    rental_file.close
+  end
```

Comment:
- [ ] Currently, your app saves the data to the files. However, you are expected to also handle scenarios when one or more storage JSON files are not present. When any file is not present, you are expected to create the file as well.

  For example, if I remove the JSON files and run the app, it throws errors, but it is expected that such scenarios are handled in some way or those files are created if they are absent. Kindly implement that.

  ![rental-json](https://user-images.githubusercontent.com/54848363/181308516-32df5789-713e-43bd-b250-3ab7f88ff4ab.jpg)

<hr />
-  [ ] Additionally, when the files are present but empty, it throws errors as well. I think to resolve this scenario you can add an empty array if the file is empty, depending on your code.

  ![empty](https://user-images.githubusercontent.com/54848363/181309267-560e2558-5d5b-46bf-8c76-0dcde22042a9.jpg)

Reply:
Done 👍🏽